### PR TITLE
Review JavaScript connection close

### LINKED ISSUE #3105: Review JavaScript connection close
https://github.com/zeroc-ice/ice/blob/34ce55cb67eee969459b618bfe7e9a5cab91e2c9/js/src/Ice/ConnectionI.js#L919

In most implementation we close the transceiver in the Closed state, but JavaScript does in the Finished stated, this difference deserves a comment at the very least. Not clear why we cannot do in the closed state.

### PR DESCRIPTION
**Pull request by pepone**
Fix #3105

### CODE REVIEW

File: js/src/Ice/ConnectionI.js [hunk 1]
```diff
@@ -915,13 +915,13 @@ export class ConnectionI {
                     }
                     this._batchRequestQueue.destroy(this._exception);
                     this._transceiver.unregister();
+                    this._transceiver.close();
```
Comment (pepone) [inline js/src/Ice/ConnectionI.js:918]: This matches what we do in other implementations, close the socket when we enter StateClosed, and destroy it when we enter StateFinished.

File: js/src/Ice/TcpTransceiver.js [hunk 2]
```diff
@@ -122,6 +122,19 @@ if (typeof net.createConnection === "function") {
                 return;
             }
 
+            try {
+                this._fd.end();
```
Comment (pepone) [inline js/src/Ice/TcpTransceiver.js:126]: sends a FIN packet, see https://nodejs.org/api/net.html#socketenddata-encoding-callback

File: js/src/Ice/WSTransceiver.js [hunk 3]
```diff
@@ -133,6 +133,9 @@ if (typeof WebSocket !== "undefined") {
             }
         }
 
+        destroy() {
+        }
```
Comment (pepone) [inline js/src/Ice/WSTransceiver.js:137]: No destroy for WebSocket, there is only close.

File: js/src/Ice/TcpTransceiver.js (hunk 2 above)
Comment (externl) [reply js/src/Ice/TcpTransceiver.js:126]: What exceptions can this function throw?

**Review by externl — APPROVED** [review]

File: js/src/Ice/WSTransceiver.js [hunk 4]
```diff
@@ -133,6 +133,8 @@ if (typeof WebSocket !== "undefined") {
             }
         }
 
+        destroy() {}
```
[the remark this answers is not in the document]
Comment (pepone) [reply js/src/Ice/WSTransceiver.js:136]: There is no destroy method for WebSocket, just close which was already called on close.

File: js/src/Ice/TcpTransceiver.js (hunk 2 above)
Comment (pepone) [reply js/src/Ice/TcpTransceiver.js]: You are right, no need for try/catch `.end()` doesn't throw

**Pull request merged**